plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/GlueMetastoreMethod.java

```diff
@@ -0,0 +1,90 @@
+package io.trino.plugin.iceberg;
+
+import io.trino.plugin.hive.metastore.glue.GlueMetastoreApiStats;
+import io.trino.plugin.hive.metastore.glue.GlueMetastoreStats;
+
+public enum GlueMetastoreMethod
+{
+    BATCH_CREATE_PARTITION,
+    BATCH_UPDATE_PARTITION,
+    CREATE_DATABASE,
+    CREATE_PARTITIONS,
+    CREATE_TABLE,
+    DELETE_COLUMN_STATISTICS_FOR_PARTITION,
+    DELETE_COLUMN_STATISTICS_FOR_TABLE,
+    DELETE_DATA_BASE,
+    DELETE_PARTITION,
+    DELETE_TABLE,
+    GET_COLUMN_STATISTICS_FOR_PARTITION,
+    GET_COLUMN_STATISTICS_FOR_TABLE,
+    GET_DATABASE,
+    GET_DATABASES,
+    GET_PARTITION,
+    GET_PARTITION_BY_NAME,
+    GET_PARTITION_NAMES,
+    GET_PARTITIONS,
+    GET_TABLE,
+    GET_TABLES,
+    UPDATE_COLUMN_STATISTICS_FOR_PARTITION,
+    UPDATE_COLUMN_STATISTICS_FOR_TABLE,
+    UPDATE_DATABASE,
+    UPDATE_PARTITION,
+    UPDATE_TABLE;
+
+    public GlueMetastoreApiStats getStatFrom(GlueMetastoreStats stats) {
+        switch (this) {
+            case BATCH_CREATE_PARTITION:
+                return stats.getBatchCreatePartition();
+            case BATCH_UPDATE_PARTITION:
+                return stats.getBatchUpdatePartition();
```

Comment:
You can define these inline on the enum values itself

```java
UPDATE_TABLE(GlueMetastoreStats::getUpdateTable),
/**/;

private final Function<GlueMetastoreStats, GlueMetastoreApiStats> extractor;

GlueMetastoreMethod(Function<GlueMetastoreStats, GlueMetastoreApiStats> extractor)
{
    this.extractor = requireNonNull(extractor, "extractor is null");
}
```

Reply:
ah this is much better. My brain was too much on Rust mode.